### PR TITLE
added support for RGBA images

### LINKED ISSUE #104029: SummaryWriter.add_embedding not working for RGBA images
### 🐛 Describe the bug

The SummaryWriter.add_embedding does not work for images with transparency (RGBA), although tensorboard's embedding projector does support this.

`import numpy as np
import torch
from torch.utils.tensorboard import SummaryWriter

writer = SummaryWriter('embedding_logs')

N_samples = 6
N_features = 24
N_channels = 3
dim_img =(128,128)

embeddings = np.random.random((N_samples,N_features))
imgs1 = np.random.random(size=(N_samples,N_channels,dim_img[0],dim_img[1]))
print(imgs1.shape) # (6, 3, 128, 128)

# Working
writer.add_embedding(
    embeddings, 
    label_img = torch.tensor(imgs1),
    global_step='RGB'
)

mask = np.random.randint(low=0, high=1, size=(N_samples,1,dim_img[0],dim_img[1]))
imgs2 = np.concatenate((imgs1, mask,), axis=1)
print(imgs2.shape) # (6, 4, 128, 128)

# Not Working
writer.add_embedding(
    embeddings, 
    label_img = torch.tensor(imgs2),
    global_step='RGBA'
)`

Error Message:
`Traceback (most recent call last):
  File "/Users/alex5753/Documents/Python/pytorch/test_embedding_projector.py", line 28, in <module>
    writer.add_embedding(
  File "/Users/alex5753/Documents/Python/pytorch/.env_torch/lib/python3.10/site-packages/torch/utils/tensorboard/writer.py", line 949, in add_embedding
    make_sprite(label_img, save_path)
  File "/Users/alex5753/Documents/Python/pytorch/.env_torch/lib/python3.10/site-packages/torch/utils/tensorboard/_embedding.py", line 45, in make_sprite
    arranged_img_CHW = make_grid(make_np(label_img), ncols=nrow)
  File "/Users/alex5753/Documents/Python/pytorch/.env_torch/lib/python3.10/site-packages/torch/utils/tensorboard/_utils.py", line 74, in make_grid
    assert I.ndim == 4 and I.shape[1] == 3
AssertionError`

The error occur because the functionality in `torch/utils/tensorboard/_embedding.py, make_sprite()` and `torch/utils/tensorboard/_utils.py, make_grid` have been hard coded to only support three image channels (RGB). Can probably be fixed by reading the amount of channels from the image directly.

### Versions

Collecting environment information...
PyTorch version: 2.0.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.4 (x86_64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.8 (v3.10.8:aaaf517424, Oct 11 2022, 10:14:40) [Clang 13.0.0 (clang-1300.0.29.30)] (64-bit runtime)
Python platform: macOS-13.4-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz

Versions of relevant libraries:
[pip3] numpy==1.25.0
[pip3] torch==2.0.1
[pip3] torchvision==0.15.2
[conda] Could not collect

### PR DESCRIPTION
**Pull request by Leminen**
Fixes #104029

### CODE REVIEW

File: torch/utils/tensorboard/_utils.py [hunk 1]
```diff
@@ -71,13 +71,14 @@ def make_grid(I, ncols=8):
     assert isinstance(I, np.ndarray), "plugin error, should pass numpy array here"
     if I.shape[1] == 1:
         I = np.concatenate([I, I, I], 1)
-    assert I.ndim == 4 and I.shape[1] == 3
+    assert I.ndim == 4 and (I.shape[1] == 3 or I.shape[1] == 4)
     nimg = I.shape[0]
```
Comment (vadimkantorov) [inline torch/utils/tensorboard/_utils.py:75]: could probably be improved as `nimg, C, H, W = I.shape`

**Review by jbschlosser — COMMENTED** [review]
Thanks for the contribution! Could you please add some tests for this case in `test/test_tensorboard.py`?

**Pull request closed without merging**